pkg/neorpc/rpcevent/filter.go

```diff
@@ -66,7 +67,28 @@ func Matches(f Comparator, r Container) bool {
 		notification := r.EventPayload().(*state.ContainedNotificationEvent)
 		hashOk := filt.Contract == nil || notification.ScriptHash.Equals(*filt.Contract)
 		nameOk := filt.Name == nil || notification.Name == *filt.Name
-		return hashOk && nameOk
+		parametersOk := true
+		if len(filt.Parameters) > 0 {
+			stackItems := notification.Item.Value().([]stackitem.Item)
+			parameters, err := filt.ParametersSI()
+			if err != nil {
+				return false
+			}
+			for i, p := range parameters {
+				if p.Type() == stackitem.AnyT && p.Value() == nil {
+					continue
+				}
+				if i >= len(stackItems) {
+					parametersOk = false
+					break
+				}
```

Comment:
Move length check out of the cycle, fail fast if violated.

Reply:
oh, right, now with `ParametersSI` it is possible, fixed. however it still had some part of my original idea: i do not think [`Integer`, `Any`, `Any`, `Any`, `Any`, `Any`, `Any`, `Any`, `Any`, `Any`, `Any`, `Any`] is a bad filter for a notification that has only 1 stackitem, cause it literally says that it does not carry about 2th+ param and the previous code allowed it
but again, i think i agree with you more, fixed

Reply:
> it does not carry about 2th+ param

Such filter explicitly says that it requires 2+ parameters to be present in the original notification. If they are not present, then notification doesn't match filter. And we don't care that corresponding 2+ filters are `Any`.